tests/unit/pidfile/pidfile_test.c

```diff
@@ -29,175 +30,175 @@ MTF_BEGIN_UTEST_COLLECTION_PRE(pidfile_test, collection_pre)
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_missing_socket_path)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-missing-socket-path", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_root_wrong_type)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-root-wrong-type", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_alias_wrong_type)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-alias-wrong-type", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_pid_wrong_type)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-pid-wrong-type", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_socket_wrong_type)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-socket-wrong-type", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_socket_path_wrong_type)
 {
-    int rc;
+    merr_t err;
     char home[PATH_MAX];
     struct pidfile content;
 
     snprintf(home, sizeof(home), "%s/deserialize-socket-path-wrong-type", config_root);
 
-    rc = pidfile_deserialize(home, &content);
-    ASSERT_EQ(EINVAL, rc);
+    err = pidfile_deserialize(home, &content);
+    ASSERT_EQ(EINVAL, merr_errno(err));
 }
 
 MTF_DEFINE_UTEST(pidfile_test, deserialize_home_dne)
 {
-    int rc;
+    merr_t err;
     struct pidfile content;
 
-    rc = pidfile_deserialize("/this-does-not-exist", &content);
-    ASSERT_EQ(ENOENT, rc);
+    err = pidfile_deserialize("/this-does-not-exist", &content);
+    ASSERT_EQ(ENOENT, merr_errno(err));
```

Comment:
I see...  So technically you should use merr() to mint a merr_t, is this a common thing now?  It doesn't help us much to use merr_t to carry around simple errnos, esp in a case like this where the func sig now lies about what it is returning.  Are you planning to change the pidfile stuff to return actual merr_t's rather than errno?